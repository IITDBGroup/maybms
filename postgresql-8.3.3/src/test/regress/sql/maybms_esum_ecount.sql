--test for esum

create table s (a int, b int);
insert into s values (1,1), (1,2),(1,3),(2,1),(2,2),(2,3);

create table r as repair key a in s weight by b;

select a, esum( b ) from r group by a; 

select esum( b ) from r; 

select a, ecount( b ) from r group by a; 

select ecount( b ) from r; 

--test for ecount without arguments, the results should be the same as those 
--with any other argument

select a, ecount() from r group by a; 

select ecount() from r; 

drop table r;
drop table s;
