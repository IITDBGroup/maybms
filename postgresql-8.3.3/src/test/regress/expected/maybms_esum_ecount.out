--test for esum
create table s (a int, b int);
insert into s values (1,1), (1,2),(1,3),(2,1),(2,2),(2,3);
create table r as repair key a in s weight by b;
select a, esum( b ) from r group by a; 
 a |  esum   
---+---------
 2 | 2.33333
 1 | 2.33333
(2 rows)

select esum( b ) from r; 
  esum   
---------
 4.66667
(1 row)

select a, ecount( b ) from r group by a; 
 a | ecount 
---+--------
 2 |      1
 1 |      1
(2 rows)

select ecount( b ) from r; 
 ecount 
--------
      2
(1 row)

--test for ecount without arguments, the results should be the same as those 
--with any other argument
select a, ecount() from r group by a; 
 a | ecount 
---+--------
 2 |      1
 1 |      1
(2 rows)

select ecount() from r; 
 ecount 
--------
      2
(1 row)

drop table r;
drop table s;
